packages/hydra-cli/test/helpers/VariantsRenderer.test.ts

```diff
@@ -137,17 +137,17 @@ describe('VariantsRenderer', () => {
     const gen = new VariantsRenderer(model)
     const rendered = c(gen.render(variantsTemplate))
     expect(rendered).to.include(
-      c(`@Field(() => String, { nullable: true, }) hates?: string`),
+      c(`@StringField({ nullable: true, }) hates?: string;`),
       'should render nullable fields'
     )
 
     expect(rendered).to.include(
-      c(`@Field(() => [String]!, {}) loves!: string[]`),
+      c(` @StringField({}) loves!: string[];`),
```

Comment:
I think this brings back an issue that was solved in the past w.r.t. arrays inside variants: https://github.com/Joystream/hydra/pull/429.
Now if you add a variant like:
```
type SomeVariant @variant {
  listOfStrings: [String]
}
```

The output type is:
```
type SomeVariant {
  listOfStrings: String
}
```

Which is suboptimal.

Probably what you want to look for instead is fixing the [`withTsTypeAndDecorator`](https://github.com/Joystream/hydra/blob/master/packages/hydra-cli/src/generate/field-context.ts#L57) function so that it returns `BigInt` in case the `f.type` is `BigInt`, and `[BigInt]` in case `f.isArray() && f.type === 'BigInt'`. I'm just guessing that something like this will work, but there is probaby a better way to express it in the code.

Reply:
This is probably a bit more complex, since for `BigNumber`s normally the `@NumericField` warthog decorator is used.
The simplest solution I can think of is:
- If `f.type` is `BigInt` (but is not array) render it with `@NumericField` decorator (just like in this PR)
- If it's both `BigInt` and an array - throw an error saying it's not supported (unless you find an easy way to add support for this case)
- Otherwise render field the "old way" (so we preserve arrays of `String`s, `Int`s etc. inside variants)

Reply:
I looked at the `model.ts.mst` template file for reference and turns out that, it uses two separate decorators([array](https://github.com/zeeshanakram3/hydra/blob/6f90415780b93078b92c9cd2d345ce48482b8b94/packages/hydra-cli/src/templates/entities/model.ts.mst#L126), [scalar](https://github.com/zeeshanakram3/hydra/blob/6f90415780b93078b92c9cd2d345ce48482b8b94/packages/hydra-cli/src/templates/entities/model.ts.mst#L126)) for `Scalar` & `Array` fields, For testing purposes, I tried the same approach in the variant template file also. 

This approach fixed the problem of no array support inside the variant. All of the integration tests also passed after introducing this change, do you think this approach will work? OR does it also introduces some unknown problems?

Reply:
So I found 2 issues with this approach:
- `src/modules/variants/variants.model.ts(441,4): error TS2304: Cannot find name 'CustomField'.` during `warthog codegen` step in the Joystream monorepo
- Something I overlooked before: It seems like after this change some variants that previously didn't have a `WhereInput` (like `GeographicalAreaCountry`), now have this type generated, although it's probably not a big issue.


However I decided to test the straightforward solution, in which I just added `BigInt` import from `@joystream/warthog` in `variants.mst` and in the `constants.ts` I changed `numeric.gqlType` to `BigInt`.

This solution doesn't seem to have any unwanted side-effects and I also ran a few tests in Joystream monorepo with thoses changes and they all passed.

Additionally I introduced a test `someStuff` field in `CategoryStatusArchived` of type `[BigInt!]` and tested a few other `BigInt` fields inside variants through a query: 
![image](https://user-images.githubusercontent.com/19978260/217263207-f6803ce4-1768-4fef-9df0-eab1952edcbf.png)

The solution appears to work correctly even for very high numbers (ie. `1000000000000000000000000000000` inside `TransactionalStatusBuyNow`.

What do you think?

Reply:
Changes compared to current PR: https://github.com/Lezek123/hydra/commit/2d1425be5583c154e8833153e0d56c49704880bf

Reply:
> However I decided to test the straightforward solution, in which I just added BigInt import from @joystream/warthog in variants.mst and in the constants.ts I changed numeric.gqlType to BigInt.

I agree this is much simpler solution, I have implemented this way. Thanks for evaluating all options  :pray: